src/ukify/ukify.py

```diff
@@ -878,24 +931,27 @@ def merge_sbat(input_pe: list[Path], input_text: list[str]) -> str:
 """
 
 
-def make_uki(opts: argparse.Namespace) -> None:
+def make_uki(opts: UkifyConfig) -> None:
+    assert opts.output is not None
+
     # kernel payload signing
 
     sign_args_present = opts.sb_key or opts.sb_cert_name
     sign_kernel = opts.sign_kernel
     linux = opts.linux
 
     if sign_args_present:
+        assert opts.linux is not None
         assert opts.signtool is not None
 
-        if not sign_kernel and opts.linux is not None:
+        if not sign_kernel:
```

Comment:
Also this one. I'd like to make the last commit is purely for replacing argparse.Namespace with UkifyConfig, without changing any logic. Though, not a strong request.

Reply:
For this and the one above I'd like to keep them in the commit adding `UkifyConfig` to keep it atomic in the sense that all necessary changes to accommodate it, land at the same time, because otherwise mypy might complain about it as unnecessary beforehand. I haven't checked the combinations of what could be mad earlier, but it's also a bit deus ex machina without it, since the way to find the changes was tying down the input.

Reply:
That makes sense.